python/ml4ir/base/model/relevance_model.py

```diff
@@ -350,6 +357,28 @@ def from_univariate_interaction_model(
             logger=logger,
         )
 
+    def add_scheduler_as_callback(self, monitor_metric, model_config):
+        """Adding reduce lr on plateau as a callback if specified"""
+        if model_config and 'lr_schedule' in model_config:
+            lr_schedule = model_config['lr_schedule']
+            lr_schedule_key = lr_schedule['key']
+            if lr_schedule_key == LearningRateScheduleKey.REDUCE_LR_ON_PLATEAU:
+                if monitor_metric is None:
+                    reduce_lr = tf.keras.callbacks.ReduceLROnPlateau(factor=lr_schedule.get('factor', 0.5),
+                                                                     patience=lr_schedule.get('patience', 1),
+                                                                     min_lr=lr_schedule.get('min_lr', 0.0001),
+                                                                     mode=lr_schedule.get('mode', 'auto'),
+                                                                     verbose=1)
+                elif not monitor_metric.startswith("val_"):
```

Comment:
This is a personal taste comment; using if/elif in this way (without else) leaves the reader wondering if anything is missing. Here, we coud have 
```python
if monitor_metric is None:
  return tf.keras...()
if not monitor_metric.startswith("val_"):
  return tf.keras...()
```
which still makes me wondering if we are missing the case that the user acutally passes a valid monitor metric. it seems like passing `val_accuracy` is not treated here. 
- Are we treating the case where the user passes a valid monitoring metric?

Reply:
I'm using the passed metric, it should be fine now.